client/consensus/slots/src/lib.rs

```diff
@@ -149,17 +149,23 @@ pub trait SimpleSlotWorker<B: BlockT> {
 	) -> Box<
 		dyn Fn(
```

Comment:
Why not just make the function async?

Reply:
Fair question. I was just following the existing pattern, but I indeed didn't find a reason for that and changed last PR to make it just an async method. `dyn Fn()` itself can't be defined as async in Rust AFAIK.